pkg/config/config.go

```diff
@@ -150,12 +145,36 @@ func (gc *Config) ShouldSkipTLSForAuthority(candidateAuthority string) bool {
 }
 
 func (gc *Config) addTrustedCerts(certChain string) (err error) {
+	backupCertsFilePath := backupSystemCertsFile
+	systemCertsFilePath := systemCertsFile
+
+	if gc.BackupCaBundlePath != "" && gc.SystemCaBundlePath != "" {
+		backupCertsFilePath = gc.BackupCaBundlePath
+		systemCertsFilePath = gc.SystemCaBundlePath
+	}
+
+	src, err := os.OpenFile(backupCertsFilePath, os.O_RDONLY, os.ModeExclusive)
+	if err != nil {
+		return fmt.Errorf("Opening original certs file: %s", err)
+	}
+	defer src.Close()
+
+	dst, err := os.OpenFile(systemCertsFilePath, os.O_CREATE|os.O_RDWR|os.O_TRUNC, os.ModeExclusive)
+	if err != nil {
+		return fmt.Errorf("Opening certs file: %s", err)
+	}
+	defer dst.Close()
```

Comment:
we should be checking on close errors. flushing might happen in close for writes.